1436790/1438462: Part 5: 20 files with MSDN link fixes.

DennisLee-DennisLee opened the pull request:
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

DennisLee-DennisLee committed:
Fixed 20 files with MSDN links.

mairaw commented on docs/visual-basic/programming-guide/language-features/variables/how-to-make-an-object-variable-not-refer-to-any-instance.md:
```diff
@@ -31,4 +31,4 @@ You can disassociate an object variable from any object instance by setting it t
 - [Nothing](../../../../visual-basic/language-reference/nothing.md)
 - [Try...Catch...Finally Statement](../../../../visual-basic/language-reference/statements/try-catch-finally-statement.md)
-- [Troubleshooting Exceptions: System.NullReferenceException](https://msdn.microsoft.com/library/4822b0b4-8105-43fb-887a-3cc51ff02899)
+- [Troubleshooting Exceptions: System.NullReferenceException](https://docs.microsoft.com/previous-versions/sxw2ez55(v=vs.140))
```
let's remove this one too

```suggestion
```

mairaw commented on docs/visual-basic/programming-guide/language-features/variables/variable-declaration.md:
````diff
@@ -55,7 +55,7 @@ Dim l, m As Long, x, y As Single
 ```  
   
- For more information on data types, see [Data Types](../../../../visual-basic/programming-guide/language-features/data-types/index.md). For more information on objects, see [Objects and Classes](../../../../visual-basic/programming-guide/language-features/objects-and-classes/index.md) and [Programming with Components](https://msdn.microsoft.com/library/d4d4fcb4-e0b8-46b3-b679-7ee0026eb9e3).  
+ For more information on data types, see [Data Types](../../../../visual-basic/programming-guide/language-features/data-types/index.md). For more information on objects, see [Objects and Classes](../../../../visual-basic/programming-guide/language-features/objects-and-classes/index.md) and [Programming with Components](https://docs.microsoft.com/previous-versions/0ffkdtkf(v=vs.140)).  
````
```suggestion
 For more information on data types, see [Data Types](../../../../visual-basic/programming-guide/language-features/data-types/index.md). For more information on objects, see [Objects and Classes](../../../../visual-basic/programming-guide/language-features/objects-and-classes/index.md) and [Programming with Components](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2013/0ffkdtkf(v=vs.120)).  
```

mairaw commented on docs/visual-basic/reference/command-line-compiler/compiler-options-listed-alphabetically.md:
```diff
@@ -76,6 +76,5 @@ The Visual Basic command-line compiler is provided as an alternative to compilin
   
 ## See also
 - [Visual Basic Compiler Options Listed by Category](../../../visual-basic/reference/command-line-compiler/compiler-options-listed-by-category.md)
-- [Introduction to the Project Designer](https://msdn.microsoft.com/library/898dd854-c98d-430c-ba1b-a913ce3c73d7)
```
Perhaps link to this one instead?
https://docs.microsoft.com/en-us/visualstudio/ide/managing-project-and-solution-properties?view=vs-2017

Also can you remove the following two C# links from here too?

mairaw commented on docs/visual-basic/reference/command-line-compiler/compiler-options-listed-by-category.md:
```diff
@@ -122,6 +122,6 @@ The Visual Basic command-line compiler is provided as an alternative to compilin
   
 ## See also
 - [Visual Basic Compiler Options Listed Alphabetically](../../../visual-basic/reference/command-line-compiler/compiler-options-listed-alphabetically.md)
-- [Introduction to the Project Designer](https://msdn.microsoft.com/library/898dd854-c98d-430c-ba1b-a913ce3c73d7(v=vs.100))
```
Same here

mairaw commented on docs/visual-basic/reference/net-framework-reference-information.md:
```diff
@@ -23,7 +23,7 @@ This topic provides links to information about how to work with the [!INCLUDE[dn
  [Tools](../../framework/tools/index.md)  
  Describes the tools that you can use to develop, configure, and deploy applications by using .NET Framework technologies.  
   
- [.NET Framework Samples](https://msdn.microsoft.com/library/177055f8-4a1f-43e7-aee6-995c196079b1)  
```
Remove this link instead

mairaw commented on docs/visual-basic/walkthroughs.md:
```diff
@@ -41,7 +41,7 @@ Walkthroughs give step-by-step instructions for common scenarios, which makes th
  [Implementing Inheritance with COM Objects](../visual-basic/programming-guide/com-interop/walkthrough-implementing-inheritance-with-com-objects.md)  
  Demonstrates how to use Visual Basic 6.0 to create a COM object containing a class, and then use it as a base class in Visual Basic.  
   
- [Multithreading](https://msdn.microsoft.com/library/2cbf5116-8499-4af9-818c-6f7c1c2ad2c9)  
```
remove this link instead

mairaw commented on docs/visual-basic/walkthroughs.md:
```diff
@@ -74,5 +74,5 @@ Walkthroughs give step-by-step instructions for common scenarios, which makes th
  Demonstrates how to create a simple object and persist its data to a file.  
   
- [Test-First Support with the Generate from Usage Feature](https://msdn.microsoft.com/library/764c17a4-cd95-4c23-bf63-d92d9c5adfb2)  
+ [Test-First Support with the Generate from Usage Feature](/visualstudio/ide/walkthrough-test-first-support-with-the-generate-from-usage-feature)  
```
```suggestion
 [Walkthrough: Test-First Support with the Generate From Usage Feature](/visualstudio/ide/walkthrough-test-first-support-with-the-generate-from-usage-feature)  
```

mairaw commented on includes/migration-guide/retargeting/core/changes-path-normalization.md:
```diff
@@ -2,7 +2,7 @@
 |   |   |
 |---|---|
-|Details|Starting with apps that target the .NET Framework 4.6.2, the way in which the runtime normalizes paths has changed.Normalizing a path involves modifying the string that identifies a path or file so that it conforms to a valid path on the target operating system. Normalization typically involves:<ul><li>Canonicalizing component and directory separators.</li><li>Applying the current direct ...
+|Details|Starting with apps that target the .NET Framework 4.6.2, the way in which the runtime normalizes paths has changed.Normalizing a path involves modifying the string that identifies a path or file so that it conforms to a valid path on the target operating system. Normalization typically involves:<ul><li>Canonicalizing component and directory separators.</li><li>Applying the current direct ...
```
@rpetrusha FYI this impacts the app-compat includes. Since this content from the folder includes/migration-guide is also used somewhere else, let's add the domain on these 

```suggestion
|Details|Starting with apps that target the .NET Framework 4.6.2, the way in which the runtime normalizes paths has changed.Normalizing a path involves modifying the string that identifies a path or file so that it conforms to a valid path on the target operating system. Normalization typically involves:<ul><li>Canonicalizing component and directory separators.</li><li>Applying the current directory to a relative path.</li><li>Evaluating the relative directory (.) or the parent directory (..) in a path.</li><li>Trimming specified characters.</li></ul>Starting with apps that target the .NET Framework 4.6.2, the following changes in path normalization are enabled by default:<ul><li>The runtime defers to the operating system's [GetFullPathName](https://docs.microsoft.com/windows/desktop/api/fileapi/nf-fileapi-getfullpathnamea) function to normalize paths.</li><li>Normalization no longer involves trimming the end of directory segments (such as a space at the end of a directory name).</li><li>Support for device path syntax in full trust, including `\\.\` and, for file I/O APIs in mscorlib.dll, `\\?\`.</li><li>The runtime does not validate device syntax paths.</li><li>The use of device syntax to access alternate data streams is supported.</li></ul>These changes improve performance while allowing methods to access previously inaccessible paths. Apps that target the .NET Framework 4.6.1 and earlier versions but are running under the .NET Framework 4.6.2 or later are unaffected by this change.|
```

mairaw commented on includes/migration-guide/retargeting/networking/default-value-servicepointmanagersecurityprotocol.md:
```diff
@@ -2,7 +2,7 @@
 |   |   |
 |---|---|
-|Details|Starting with apps that target the .NET Framework 4.7, the default value of the <xref:System.Net.ServicePointManager.SecurityProtocol?displayProperty=nameWithType> property is <xref:System.Net.SecurityProtocolType.SystemDefault?displayProperty=nameWithType>. This change allows .NET Framework networking APIs based on SslStream (such as FTP, HTTPS, and SMTP) to inherit the default security ...
+|Details|Starting with apps that target the .NET Framework 4.7, the default value of the <xref:System.Net.ServicePointManager.SecurityProtocol?displayProperty=nameWithType> property is <xref:System.Net.SecurityProtocolType.SystemDefault?displayProperty=nameWithType>. This change allows .NET Framework networking APIs based on SslStream (such as FTP, HTTPS, and SMTP) to inherit the default security ...
```
```suggestion
|Details|Starting with apps that target the .NET Framework 4.7, the default value of the <xref:System.Net.ServicePointManager.SecurityProtocol?displayProperty=nameWithType> property is <xref:System.Net.SecurityProtocolType.SystemDefault?displayProperty=nameWithType>. This change allows .NET Framework networking APIs based on SslStream (such as FTP, HTTPS, and SMTP) to inherit the default security protocols from the operating system instead of using hard-coded values defined by the .NET Framework. The default varies by operating system and any custom configuration performed by the system administrator. For information on the default SChannel protocol in each version of the Windows operating system, see [Protocols in TLS/SSL (Schannel SSP)](https://docs.microsoft.com/windows/desktop/SecAuthN/protocols-in-tls-ssl--schannel-ssp-).</p>For applications that target an earlier version of the .NET Framework, the default value of the <xref:System.Net.ServicePointManager.SecurityProtocol?displayProperty=nameWithType> property depends on the version of the .NET Framework targeted. See the [Networking section of Retargeting Changes for Migration from .NET Framework 4.5.2 to 4.6](~/docs/framework/migration-guide/retargeting/4.5.2-4.6.md#networking) for more information.|
```

mairaw commented on includes/migration-guide/retargeting/networking/sslstream-supports-tls-alerts.md:
```diff
@@ -2,7 +2,7 @@
 |   |   |
 |---|---|
-|Details|After a failed TLS handshake, an <xref:System.IO.IOException?displayProperty=name> with an inner <xref:System.ComponentModel.Win32Exception?displayProperty=name> exception will be thrown by the first I/O Read/Write operation. The <xref:System.ComponentModel.Win32Exception.NativeErrorCode?displayProperty=name> code for the <xref:System.ComponentModel.Win32Exception?displayProperty=name> c ...
+|Details|After a failed TLS handshake, an <xref:System.IO.IOException?displayProperty=name> with an inner <xref:System.ComponentModel.Win32Exception?displayProperty=name> exception will be thrown by the first I/O Read/Write operation. The <xref:System.ComponentModel.Win32Exception.NativeErrorCode?displayProperty=name> code for the <xref:System.ComponentModel.Win32Exception?displayProperty=name> c ...
```
```suggestion
|Details|After a failed TLS handshake, an <xref:System.IO.IOException?displayProperty=name> with an inner <xref:System.ComponentModel.Win32Exception?displayProperty=name> exception will be thrown by the first I/O Read/Write operation. The <xref:System.ComponentModel.Win32Exception.NativeErrorCode?displayProperty=name> code for the <xref:System.ComponentModel.Win32Exception?displayProperty=name> can be mapped to the TLS Alert from the remote party using the [Schannel error codes for TLS and SSL alerts](https://docs.microsoft.com/windows/desktop/SecAuthN/schannel-error-codes-for-tls-and-ssl-alerts). For more information, see [RFC 2246: Section 7.2.2 Error alerts](https://tools.ietf.org/html/rfc2246#section-7.2.2). <br/>The behavior in .NET Framework 4.6.2 and earlier is that the transport channel (usually TCP connection) will timeout during either Write or Read if the other party failed the handshake and immediately afterwards rejected the connection.|
```

mairaw commented on includes/migration-guide/retargeting/networking/tls-1x-by-default-passes-schsendauxrecord-flag-underlying-schannel-api.md:
```diff
@@ -2,8 +2,8 @@
 |---|---|
-|Details|When using TLS 1.x, the .NET Framework relies on the underlying Windows SCHANNEL API. Starting with .NET Framework 4.6, the [SCH_SEND_AUX_RECORD](https://msdn.microsoft.com/library/windows/desktop/aa379810.aspx) flag is passed by default to SCHANNEL. This causes SCHANNEL to split data to be encrypted into two separate records, the first as a single byte and the second as <em>n</em>-1 byt ...
-|Suggestion|If this change breaks communication with an existing server, you can disable sending the [SCH_SEND_AUX_RECORD](https://msdn.microsoft.com/library/windows/desktop/aa379810.aspx) flag and restore the previous behavior of not splitting data into separate records by adding the following switch to the [<AppContextSwitchOverrides>](~/docs/framework/configure-apps/file-schema/runtime/appcont ...
+|Details|When using TLS 1.x, the .NET Framework relies on the underlying Windows SCHANNEL API. Starting with .NET Framework 4.6, the [SCH_SEND_AUX_RECORD](/windows/desktop/api/schannel/ns-schannel-_schannel_cred) flag is passed by default to SCHANNEL. This causes SCHANNEL to split data to be encrypted into two separate records, the first as a single byte and the second as <em>n</em>-1 bytes.In ra ...
```
```suggestion
|Details|When using TLS 1.x, the .NET Framework relies on the underlying Windows SCHANNEL API. Starting with .NET Framework 4.6, the [SCH_SEND_AUX_RECORD](https://docs.microsoft.com/windows/desktop/api/schannel/ns-schannel-_schannel_cred) flag is passed by default to SCHANNEL. This causes SCHANNEL to split data to be encrypted into two separate records, the first as a single byte and the second as <em>n</em>-1 bytes. In rare cases, this breaks communication between clients and existing servers that make the assumption that the data resides in a single record.|
```

mairaw commented on includes/migration-guide/retargeting/networking/tls-1x-by-default-passes-schsendauxrecord-flag-underlying-schannel-api.md:
```diff
@@ -2,8 +2,8 @@
-|Details|When using TLS 1.x, the .NET Framework relies on the underlying Windows SCHANNEL API. Starting with .NET Framework 4.6, the [SCH_SEND_AUX_RECORD](https://msdn.microsoft.com/library/windows/desktop/aa379810.aspx) flag is passed by default to SCHANNEL. This causes SCHANNEL to split data to be encrypted into two separate records, the first as a single byte and the second as <em>n</em>-1 byt ...
-|Suggestion|If this change breaks communication with an existing server, you can disable sending the [SCH_SEND_AUX_RECORD](https://msdn.microsoft.com/library/windows/desktop/aa379810.aspx) flag and restore the previous behavior of not splitting data into separate records by adding the following switch to the [<AppContextSwitchOverrides>](~/docs/framework/configure-apps/file-schema/runtime/appcont ...
+|Details|When using TLS 1.x, the .NET Framework relies on the underlying Windows SCHANNEL API. Starting with .NET Framework 4.6, the [SCH_SEND_AUX_RECORD](/windows/desktop/api/schannel/ns-schannel-_schannel_cred) flag is passed by default to SCHANNEL. This causes SCHANNEL to split data to be encrypted into two separate records, the first as a single byte and the second as <em>n</em>-1 bytes.In ra ...
+|Suggestion|If this change breaks communication with an existing server, you can disable sending the [SCH_SEND_AUX_RECORD](/windows/desktop/api/schannel/ns-schannel-_schannel_cred) flag and restore the previous behavior of not splitting data into separate records by adding the following switch to the [<AppContextSwitchOverrides>](~/docs/framework/configure-apps/file-schema/runtime/appcontextswitc ...
```
```suggestion
|Suggestion|If this change breaks communication with an existing server, you can disable sending the [SCH_SEND_AUX_RECORD](https://docs.microsoft.com/windows/desktop/api/schannel/ns-schannel-_schannel_cred) flag and restore the previous behavior of not splitting data into separate records by adding the following switch to the [<AppContextSwitchOverrides>](~/docs/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) element in the [<runtime>](~/docs/framework/configure-apps/file-schema/runtime/runtime-element.md) section of your app configuration file:<pre><code class="lang-xml">&lt;runtime&gt;&#13;&#10;&lt;AppContextSwitchOverrides&#13;&#10;value=&quot;Switch.System.Net.DontEnableSchSendAuxRecord=true&quot; /&gt;&#13;&#10;&lt;/runtime&gt;&#13;&#10;</code></pre> <blockquote> [!IMPORTANT] This setting is provided for backward compatibility only. Its use is otherwise not recommended.</blockquote> |
```

mairaw commented on includes/migration-guide/retargeting/security/cspparametersparentwindowhandle-now-expects-hwnd-value.md:
```diff
@@ -2,7 +2,7 @@
 |   |   |
 |---|---|
-|Details|The <xref:System.Security.Cryptography.CspParameters.ParentWindowHandle> value, introduced in .NET Framework 2.0, allows an application to register a parent window handle value such that any UI required to access the key (such as a PIN prompt or consent dialog) opens as a modal child to the specified window.Starting with apps that target the .NET Framework 4.7, a Windows Forms applicatio ...
+|Details|The <xref:System.Security.Cryptography.CspParameters.ParentWindowHandle> value, introduced in .NET Framework 2.0, allows an application to register a parent window handle value such that any UI required to access the key (such as a PIN prompt or consent dialog) opens as a modal child to the specified window.Starting with apps that target the .NET Framework 4.7, a Windows Forms applicatio ...
```
```suggestion
|Details|The <xref:System.Security.Cryptography.CspParameters.ParentWindowHandle> value, introduced in .NET Framework 2.0, allows an application to register a parent window handle value such that any UI required to access the key (such as a PIN prompt or consent dialog) opens as a modal child to the specified window.Starting with apps that target the .NET Framework 4.7, a Windows Forms application can set the <xref:System.Security.Cryptography.CspParameters.ParentWindowHandle> property with code like the following:<pre><code class="lang-csharp">cspParameters.ParentWindowHandle = form.Handle;&#13;&#10;</code></pre>In previous versions of the .NET Framework, the value was expected to be an <xref:System.IntPtr?displayProperty=name> representing a location in memory where the [HWND](https://docs.microsoft.com/windows/desktop/WinProg/windows-data-types#HWND) value resided. Setting the property to form.Handle on Windows 7 and earlier versions had no effect, but on Windows 8 and later versions, it results in a &quot;<xref:System.Security.Cryptography.CryptographicException?displayProperty=name>: The parameter is incorrect.&quot;|
```

mairaw commented on includes/migration-guide/retargeting/winforms/accessibility-improvements-windows-forms-controls.md:
```diff
@@ -3,7 +3,7 @@
 |---|---|
 |Details|Windows Forms is improving how it works with accessibility technologies to better support Windows Forms customers. These include the following changes starting with the .NET Framework 4.7.1:<ul><li>Changes to improve display during High Contrast mode.</li><li>Changes to improve the property browser experience. Property browser improvements include:</li><li>Better keyboard navigation thro ...
-|Suggestion|<strong>How to opt in or out of these changes</strong></br>In order for the application to benefit from these changes, it must run on the .NET Framework 4.7.1 or later. The application can benefit from these changes in either of the following ways:<ul><li>It is recompiled to target the .NET Framework 4.7.1. These accessibility changes are enabled by default on Windows Forms applicatio ...
+|Suggestion|<strong>How to opt in or out of these changes</strong></br>In order for the application to benefit from these changes, it must run on the .NET Framework 4.7.1 or later. The application can benefit from these changes in either of the following ways:<ul><li>It is recompiled to target the .NET Framework 4.7.1. These accessibility changes are enabled by default on Windows Forms applicatio ...
```
```suggestion
|Suggestion|<strong>How to opt in or out of these changes</strong></br>In order for the application to benefit from these changes, it must run on the .NET Framework 4.7.1 or later. The application can benefit from these changes in either of the following ways:<ul><li>It is recompiled to target the .NET Framework 4.7.1. These accessibility changes are enabled by default on Windows Forms applications that target the .NET Framework 4.7.1 or later.</li><li>It opts out of the legacy accessibility behaviors by adding the following [AppContext switch](~/docs/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) to the <code>&lt;runtime&gt;</code> section of the app.config file and setting it to <code>false</code>, as the following example shows.</li></ul><pre><code class="lang-xml">&lt;?xml version=&quot;1.0&quot; encoding=&quot;utf-8&quot;?&gt;&#13;&#10;&lt;configuration&gt;&#13;&#10;&lt;startup&gt;&#13;&#10;&lt;supportedRuntime version=&quot;v4.0&quot; sku=&quot;.NETFramework,Version=v4.7&quot;/&gt;&#13;&#10;&lt;/startup&gt;&#13;&#10;&lt;runtime&gt;&#13;&#10;&lt;!-- AppContextSwitchOverrides value attribute is in the form of &#39;key1=true/false;key2=true/false  --&gt;&#13;&#10;&lt;AppContextSwitchOverrides value=&quot;Switch.UseLegacyAccessibilityFeatures=false&quot; /&gt;&#13;&#10;&lt;/runtime&gt;&#13;&#10;&lt;/configuration&gt;&#13;&#10;</code></pre></br>Applications that target the .NET Framework 4.7.1 or later and want to preserve the legacy accessibility behavior can opt in to the use of legacy accessibility features by explicitly setting this AppContext switch to <code>true</code>.</p>For an overview of UI automation, see the [UI Automation Overview](~/docs/framework/ui-automation/ui-automation-overview.md).&lt;/p/&gt;</p><strong>Added support for UI Automation patterns and properties</strong><br/>Accessibility clients can take advantage of new WinForms accessibility functionality by using common, publicly described invocation patterns. These patterns are not WinForms-specific. For instance, accessibility clients can call the QueryInterface method on the IAccessible interface (MAAS) to obtain an IServiceProvider interface. If this interface is available, clients can use its QueryService method to request an IAccessibleEx interface. For more information, see [Using IAccessibleEx from a Client](https://docs.microsoft.com/windows/desktop/WinAuto/using-iaccessibleex-from-a-client). Starting with the .NET Framework 4.7.1, IServiceProvider and [IAccessibleEx](https://docs.microsoft.com/windows/desktop/WinAuto/iaccessibleex) (where applicable) are available for WinForms accessibility objects.</p>The .NET Framework 4.7.1 adds support for the following UI automation patterns and properties:<ul><li>The <xref:System.Windows.Forms.ToolStripSplitButton> and <xref:System.Windows.Forms.ComboBox> controls support the [Expand/Collapse pattern](~/docs/framework/ui-automation/implementing-the-ui-automation-expandcollapse-control-pattern.md).</li><li>The <xref:System.Windows.Forms.ToolStripMenuItem> control has a [ControlType](~/docs/framework/ui-automation/ui-automation-support-for-the-menubar-control-type.md) property value <xref:System.Windows.Automation.ControlType.MenuItem?displayProperty=nameWithType>.</li><li>The <xref:System.Windows.Forms.ToolStripItem> control supports the <xref:System.Windows.Automation.AutomationElement.NameProperty> property and the [Expand/Collapse pattern](~/docs/framework/ui-automation/implementing-the-ui-automation-expandcollapse-control-pattern.md).</li><li>The <xref:System.Windows.Forms.ToolStripDropDownItem> control supports <xref:System.Windows.Forms.AccessibleEvents> indicating StateChange and NameChange when drop down is expanded or collapsed.</li><li>The <xref:System.Windows.Forms.ToolStripDropDownButton> control has a [ControlType](~/docs/framework/ui-automation/ui-automation-support-for-the-menubar-control-type.md) property value of <xref:System.Windows.Automation.ControlType.MenuItem?displayProperty=nameWithType>.</li><li>The <xref:System.Windows.Forms.DataGridViewCheckBoxCell> control supports the <xref:System.Windows.Automation.TogglePattern>.</li><li>The <xref:System.Windows.Forms.NumericUpDown> and <xref:System.Windows.Forms.DomainUpDown> controls support the <xref:System.Windows.Automation.AutomationElement.NameProperty> property and have a [ControlType](~/docs/framework/ui-automation/ui-automation-support-for-the-spinner-control-type.md) of <xref:System.Windows.Automation.ControlType.Spinner?displayProperty=nameWithType>.</p></li></ul><strong>Improvements to the PropertyGrid control</strong></br>The .NET Framework 4.7.1 adds the following improvements to the PropertyBrowser control:<ul><li>The <strong>Details</strong> button in the error dialog that is displayed when the user enters an incorrect value in the <xref:System.Windows.Forms.PropertyGrid> control supports the [Expand/Collapse pattern](~/docs/framework/ui-automation/implementing-the-ui-automation-expandcollapse-control-pattern.md), state and name change notifications, and a [ControlType]~/docs/framework/ui-automation/ui-automation-support-for-the-menubar-control-type.md) property with a value of <xref:System.Windows.Automation.ControlType.MenuItem?displayProperty=nameWithType>.</li><li>The message pane displayed when the <strong>Details</strong> button of the error dialog is expanded is now keyboard accessible and allows Narrator to announce the content of the error message.</li><li>The <xref:System.Windows.Forms.AccessibleRole> of rows in the <xref:System.Windows.Forms.PropertyGrid> control have changed from &quot;Row&quot; to &quot;Cell&quot;. The cell maps to UIA ControlType &quot;DataItem&quot;, which allows it to support appropriate keyboard shortcuts and Narrator announcements.</li><li>The <xref:System.Windows.Forms.PropertyGrid> control rows that represent header items when the <xref:System.Windows.Forms.PropertyGrid> control has a <xref:System.Windows.Forms.PropertyGrid.PropertySort> property set to <xref:System.Windows.Forms.PropertySort.Categorized?displayProperty=nameWithType> have a [ControlType](~/docs/framework/ui-automation/ui-automation-support-for-the-menubar-control-type.md) property value of <xref:System.Windows.Automation.ControlType.Button?displayProperty=nameWithType>.</li><li>The <xref:System.Windows.Forms.PropertyGrid> control rows that represent header items when the <xref:System.Windows.Forms.PropertyGrid> control has a <xref:System.Windows.Forms.PropertyGrid.PropertySort>  property set to <xref:System.Windows.Forms.PropertySort.Categorized?displayProperty=nameWithType> support the [Expand/Collapse pattern](~/docs/framework/ui-automation/implementing-the-ui-automation-expandcollapse-control-pattern.md).</li><li>Improved keyboard navigation between the grid and the ToolBar above it. Pressing &quot;Shift-Tab&quot; now selects the first ToolBar button, instead of the whole ToolBar.</li><li><xref:System.Windows.Forms.PropertyGrid> controls displayed in High Contrast mode will now draw a focus rectangle around the ToolBar button which corresponds to the current <xref:System.Windows.Forms.PropertyGrid.PropertySort> property value.</li><li><xref:System.Windows.Forms.PropertyGrid> controls displayed in High Contrast mode and with a <xref:System.Windows.Forms.PropertyGrid.PropertySort> property set to <xref:System.Windows.Forms.PropertySort.Categorized?displayProperty=nameWithType> will now display the background of category headers in a highly contrasting color.</li><li><xref:System.Windows.Forms.PropertyGrid> controls better differentiates between ToolBar items with focus and the ToolBar items which indicate the current value of the <xref:System.Windows.Forms.PropertyGrid.PropertySort> property. This fix consists of a High Contrast change and a change for non-High Contrast scenarios.</li><li><xref:System.Windows.Forms.PropertyGrid> control ToolBar items which indicates the current value of the <xref:System.Windows.Forms.PropertyGrid.PropertySort> property support the <xref:System.Windows.Automation.TogglePattern>.</li><li>Improved Narrator support for distinguishing the selected alignment in the Alignment Picker.</li><li>When an empty <xref:System.Windows.Forms.PropertyGrid> control is displayed on a form, it will now receive focus where previously it would not.</p></li></ul><strong>Use of OS-defined colors in High Contrast themes</strong></br><ul><li>The <xref:System.Windows.Forms.Button> and <xref:System.Windows.Forms.CheckBox> controls with their <xref:System.Windows.Forms.ButtonBase.FlatStyle> property set to <xref:System.Windows.Forms.FlatStyle.System?displayProperty=nameWithType>, which is the default style, now use OS-defined colors in High Contrast theme when selected. Previously, text and background colors were not contrasting and were hard to read.</li><li>The <xref:System.Windows.Forms.Button>, <xref:System.Windows.Forms.CheckBox>, <xref:System.Windows.Forms.RadioButton>, <xref:System.Windows.Forms.Label>, <xref:System.Windows.Forms.LinkLabel> and <xref:System.Windows.Forms.GroupBox> controls with their <xref:System.Windows.Forms.Control.Enabled> property set to <strong>false</strong> used a shaded color to render text in High Contrast themes, resulting in low contrast against the background. Now these controls use the &quot;Disabled Text&quot; color defined by the OS. This fix applies to controls with their <code>FlatStyle</code> property set to a value other than <xref:System.Windows.Forms.FlatStyle.System?displayProperty=nameWithType>. The latter controls are rendered by the OS.</li><li><xref:System.Windows.Forms.DataGridView> now renders a visible rectangle around the content of the cell which has the current focus. Previously, this was not visible in certain High Contrast themes.</li><li><xref:System.Windows.Forms.ToolStripMenuItem> controls with their <xref:System.Windows.Forms.ToolStripMenuItem.Enabled> property set to <strong>false</strong> now use the &quot;Disabled Text&quot; color defined by the OS.</li><li><xref:System.Windows.Forms.ToolStripMenuItem> controls with their <xref:System.Windows.Forms.ToolStripMenuItem.Checked> property set to <strong>true</strong> now render the associated check mark in a contrasting system color.  Previously the check mark color was not contrasting enough and not visible in High Contrast themes.</li></ul>NOTE: Windows 10 has changed values for some high contrast system colors. Windows Forms Framework is based on the Win32 framework. For the best experience, run on the latest version of Windows and opt in to the latest OS changes by adding an app.manifest file in a test application and uncommenting the following code:<pre><code class="lang-xml">&lt;!-- Windows 10 --&gt;&#13;&#10;&lt;supportedOS Id=&quot;{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}&quot; /&gt;&#13;&#10;</code></pre></br><strong>Improved keyboard navigation</strong><ul><li>When a <xref:System.Windows.Forms.ComboBox> control has its <xref:System.Windows.Forms.ComboBox.DropDownStyle> property set to <xref:System.Windows.Forms.ComboBoxStyle.DropDownList?displayProperty=nameWithType> and is the first control in the tab order on the form, it now displays a focus rectangle when the parent form is opened using the keyboard. Before this change, keyboard focus was on this control, but a focus indicator was not rendered.</li></ul><strong>Improved Narrator support</strong><ul><li>The <xref:System.Windows.Forms.MonthCalendar> control has added support for assistive technologies to access the control, including the ability for Narrator to read the value of the control when previously it could not.</li><li>The <xref:System.Windows.Forms.CheckedListBox> control now notifies Narrator when a <xref:System.Windows.Forms.CheckBox.CheckState?displayProperty=nameWithType> property has been changed. Previously, Narrator did not receive notification and as a result users would not be informed that the <xref:System.Windows.Forms.CheckBox.CheckState> property had been updated.</li><li>The <xref:System.Windows.Forms.LinkLabel> control has changed the way it notifies Narrator of the text of in the control. Previously, Narrator announced this text twice and read &quot;&amp;&quot; symbols as real text even though they are not visible to a user. The duplicated text was removed from the Narrator announcements, as well as unnecessary &quot;&amp;&quot; symbols.</li><li>The <xref:System.Windows.Forms.DataGridViewCell> control types now correctly report the read-only status to Narrator and other assistive technologies.</li><li>Narrator is now able to read the System Menu of child windows in [Multiple-Document Interface]~/docs/framework/winforms/advanced/multiple-document-interface-mdi-applications.md) applications.</li><li>Narrator is now able to read <xref:System.Windows.Forms.ToolStripMenuItem> controls with a <xref:System.Windows.Forms.ToolStripItem.Enabled?displayProperty=nameWithType> property set to <strong>false</strong>. Previously, Narrator was unable to focus on disabled menu items to read the content.</li></ul>|
```

mairaw commented on includes/migration-guide/runtime/asp/httprequestcontentencoding-property-prohibits-utf7.md:
```diff
@@ -3,7 +3,7 @@
 |---|---|
 |Details|Beginning in .NET Framework 4.5, UTF-7 encoding is prohibited in <xref:System.Web.HttpRequest?displayProperty=name>s' bodies. Data for applications that depend on incoming UTF-7 data will not decode properly in some cases.|
-|Suggestion|Ideally, applications should be updated to not use UTF-7 encoding in <xref:System.Web.HttpRequest?displayProperty=name>s. Alternatively, legacy behavior can be restored by using the <code>aspnet:AllowUtf7RequestContentEncoding</code> attribute of the [appSettings](https://msdn.microsoft.com/library/hh975440(v=vs.110).aspx) element.|
+|Suggestion|Ideally, applications should be updated to not use UTF-7 encoding in <xref:System.Web.HttpRequest?displayProperty=name>s. Alternatively, legacy behavior can be restored by using the <code>aspnet:AllowUtf7RequestContentEncoding</code> attribute of the [appSettings](https://docs.microsoft.com/previous-versions/aspnet/hh975440(v=vs.110)) element.|
```
```suggestion
|Suggestion|Ideally, applications should be updated to not use UTF-7 encoding in <xref:System.Web.HttpRequest?displayProperty=name>s. Alternatively, legacy behavior can be restored by using the <code>aspnet:AllowUtf7RequestContentEncoding</code> attribute of the [appSettings](~/docs/framework/configure-apps/file-schema/appsettings/appsettings-element-for-configuration.md) element.|
```

mairaw commented on includes/migration-guide/runtime/data/sqlconnection-can-no-longer-connect-sql-server-1997-databases-using-via.md:
```diff
@@ -3,7 +3,7 @@
 |---|---|
 |Details|Connections to SQL Server databases using the [Virtual Interface Adapter (VIA) protocol](https://docs.microsoft.com/previous-versions/sql/sql-server-2008-r2/ms191229%28v=sql.105%29) are no longer supported. The protocol used to connect to a SQL Server database is visible in the connection string. A VIA connection will contain via:&lt;servername&gt;. If this app is connecting to SQL via a ...
-|Suggestion|The VIA protocol is deprecated, so an alternative protocol should be used to connect to SQL databases. The most common protocol used is TCP/IP. Instructions for enabling the TCP/IP protocol can be found [here](https://msdn.microsoft.com/library/bb909712.aspx). If the database is only accessed from within an intranet, the shared pipes protocol may provide better performance if the netw ...
+|Suggestion|The VIA protocol is deprecated, so an alternative protocol should be used to connect to SQL databases. The most common protocol used is TCP/IP. For more information about connecting through TCP/IP, see [Enable the TCP/IP protocol for a database instance](https://docs.microsoft.com/previous-versions/bb909712(v=vs.120)). If the database is only accessed from within an intranet, the shar ...
```
```suggestion
|Suggestion|The VIA protocol is deprecated, so an alternative protocol should be used to connect to SQL databases. The most common protocol used is TCP/IP. For more information about connecting through TCP/IP, see [Enable the TCP/IP protocol for a database instance](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2008/bb909712(v=vs.90)). If the database is only accessed from within an intranet, the shared pipes protocol may provide better performance if the network is slow.|
```

mairaw reviewed (changes requested):
Looks great @DennisLee-DennisLee. Left a few comments to be addressed before we can merge.

mairaw committed:
Apply suggestions from code review

Committing 15 of Maira's suggestions.

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>

DennisLee-DennisLee committed:
Addressed the four suggestions from Maira that needed to be done manually.

mairaw committed:
Update includes/migration-guide/retargeting/winforms/accessibility-improvements-windows-forms-controls.md

Missed this one earlier. It looks like there are two additions of "https://docs.microsoft.com" in that monster line.

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>

DennisLee-DennisLee committed:
Fixed the link text for the "Introduction to the Project Designer" link replacement.

DennisLee-DennisLee committed:
Merge branch 'v-dele-1438462-005' of https://github.com/DennisLee-DennisLee/docs into v-dele-1438462-005

DennisLee-DennisLee commented:
#sign-off

mairaw reviewed (approved):
Looks great @DennisLee-DennisLee! :shipit:
